ocp_utilities/infra.py

```diff
@@ -81,6 +85,46 @@ def assert_pods_failed_or_pending(pods):
         )
 
 
+def assert_nodes_in_healthy_condition(nodes):
+    """
+    Validates nodes are in a healthy condition.
+    Nodes Ready condition is True and none of the following node conditions is True:
+        - DiskPressure
+        - MemoryPressure
+        - PIDPressure
+        - NetworkUnavailable
+
+    Args:
+         nodes(list): List of Node objects
+
+    Raises:
+        NodesNotHealthyConditionError: if any nodes DiskPressure
+        MemoryPressure, PIDPressure, NetworkUnavailable condition is True
+    """
+    LOGGER.info("Verify all nodes are in a healthy condition")
+    unhealthy_nodes_with_conditions = {}
+    for node in nodes:
+        condition_type_list = []
+        for condition in node.instance.status.conditions:
+            if (
+                condition.type != Node.Condition.READY
```

Comment:
```
{'lastHeartbeatTime': '2023-01-31T11:00:39Z',
  'lastTransitionTime': '2023-01-31T10:58:57Z',
  'message': 'kubelet is posting ready status',
  'reason': 'KubeletReady',
  'status': 'True',
  'type': 'Ready'}
```

this is covered in `assert_nodes_ready` - I think it can be united into one function; @myakove wdyt?

Reply:
@robbiehemhub please remove `assert_nodes_ready` and test all conditions here; @dbasunag  fyi about the change

Reply:
@rnetser do we want to address this for cluster_sanity?